generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/FieldSpec.java

```diff
@@ -200,8 +195,35 @@ public FieldSpec withMustContainRestriction(MustContainRestriction mustContainRe
             this.source);
     }
 
-    public FieldSpecSource getFieldSpecSource() {
-        return this.source;
+    /** Create a predicate that returns TRUE for all (and only) values permitted by this FieldSpec */
+    public boolean permits(@NotNull Object value) {
+        if (typeRestrictions != null) {
+            if (!typeRestrictions.isTypeAllowed(IsOfTypeConstraint.Types.NUMERIC)) {
+                if (isNumeric(value)) return false;
+            }
+
+            if (!typeRestrictions.isTypeAllowed(IsOfTypeConstraint.Types.STRING)) {
+                if (isString(value)) return false;
+            }
+
+            if (!typeRestrictions.isTypeAllowed(IsOfTypeConstraint.Types.DATETIME)) {
+                if (isDateTime(value)) return false;
+            }
+        }
+
+        if (numericRestrictions != null) {
+            if (isNumeric(value) && numericRestrictions.match(value)) return false;
```

Comment:
Does "match" mean that the value is not allowed by the restriction? This is confusing

Reply:
No, I just messed up the logic D: Will fix in upcoming commit, thanks for the spot!